helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -308,16 +446,62 @@ private void batchCustomInstanceStoppableCheck(String clusterId, List<String> in
     List<String> instancesForCustomPartitionLevelChecks =
         filterInstancesForNextCheck(customInstanceLevelChecks, finalStoppableChecks);
     if (!instancesForCustomPartitionLevelChecks.isEmpty()) {
+      // add to finalStoppableChecks regardless of stoppable or not.
       Map<String, StoppableCheck> instancePartitionLevelChecks =
           performPartitionsCheck(instancesForCustomPartitionLevelChecks, restConfig,
               customPayLoads);
+      List<String> instanceForNextRound = new ArrayList<>();
```

Comment:
Still confused with next round naming. Also internally, in the batch impl, we check the isNonblocking, why we add it back?

Reply:
If the CustomStoppableCheck failed (weather it is a non blocking check or a blocking check), we will get stoppableCheck.isStoppable() to be false. The instanceForNextRound list contains instances that is stoppable (stoppableCheck.isStoppable() is true) and instances that failed nonblocking check.

Reply:
Thanks for pointing it out. The naming is confusing... I updated to a clearer one. .